src/NuGet.Core/NuGet.Packaging/NuspecReader.cs

```diff
@@ -107,81 +107,15 @@ public NuspecReader(XDocument xml, IFrameworkNameProvider frameworkProvider)
-            {
-                groupFound = true;
-
-                var groupFramework = GetAttributeValue(depGroup, TargetFramework);
-
-                var packages = new HashSet<PackageDependency>();
-
-                foreach (var depNode in depGroup.Elements(XName.Get(Dependency, ns)))
-                {
-                    VersionRange range = null;
-
-                    var rangeNode = GetAttributeValue(depNode, Version);
-
-                    if (!String.IsNullOrEmpty(rangeNode))
-                    {
-                        VersionRange.TryParse(rangeNode, out range);
-
-                        Debug.Assert(range != null, "Unable to parse range: " + rangeNode);
-                    }
-
-                    var includeFlags = GetFlags(GetAttributeValue(depNode, IncludeFlags));
-                    var excludeFlags = GetFlags(GetAttributeValue(depNode, ExcludeFlags));
-
-                    var dependency = new PackageDependency(
-                        GetAttributeValue(depNode, Id), 
-                        range,
-                        includeFlags,
-                        excludeFlags);
-
-                    packages.Add(dependency);
-                }
-
-                var framework = String.IsNullOrEmpty(groupFramework) 
-                    ? NuGetFramework.AnyFramework 
-                    : NuGetFramework.Parse(groupFramework, _frameworkProvider);
-
-                yield return new PackageDependencyGroup(framework, packages);
-            }
-
-            // legacy behavior
-            if (!groupFound)
-            {
-                var depNodes = MetadataNode.Elements(XName.Get(Dependencies, ns))
-                    .Elements(XName.Get(Dependency, ns));
-
-                var packages = new HashSet<PackageDependency>();
-
-                foreach (var depNode in depNodes)
-                {
-                    VersionRange range = null;
-
-                    var rangeNode = GetAttributeValue(depNode, Version);
-
-                    if (!String.IsNullOrEmpty(rangeNode))
-                    {
-                        VersionRange.TryParse(rangeNode, out range);
-
-                        Debug.Assert(range != null, "Unable to parse range: " + rangeNode);
-                    }
-
-                    packages.Add(new PackageDependency(GetAttributeValue(depNode, Id), range));
-                }
-
-                if (packages.Any())
-                {
-                    yield return new PackageDependencyGroup(NuGetFramework.AnyFramework, packages);
-                }
-            }
+            return GetAllDependencyGroups(useStrictVersionCheck: false);
+        }
 
-            yield break;
+        /// <summary>
+        /// Read package dependencies for all frameworks
+        /// </summary>
+        public IEnumerable<PackageDependencyGroup> GetDependencyGroups(bool useStrictVersionCheck)
+        {
+            return GetAllDependencyGroups(useStrictVersionCheck);
```

Comment:
We can keep the exact same public API service area:
1. `NuSpecReader.GetDependencyGroups()`
1. `NuSpecReader.GetDependencyGroups(bool useStrictVersionCheck)`

Number 1 will just call number two with `useStrictVersionCheck = false` right?

Optional argument is actually less preferable since this is actually a breaking change at runtime (but not at build time).